jsdoc/plugins/highcharts.jsdoc.js

```diff
@@ -1109,15 +1109,26 @@ Move them up before functional code for JSDoc to see them.`.yellow
         improveDescription({children: options});
 
         function addSeriesTypeDescription(type) {
-            var node = type;
+            var node = type,
+                capitalizedType = type.charAt(0).toUpperCase() + type.slice(1);
 
             // Make sense of the examples for general series
             if (type === 'series') {
                 type = 'line';
             }
             var s = `
 
-In TypeScript the [type](series.${type}.type) option must always be set.
+**In TypeScript:**
+- the [type](series.${type}.type) option must always be set.
+- when accessing an array of series, the combined set of all series types is
+  represented by [Highcharts.SeriesOptionsType
+  ](/class-reference/Highcharts#.SeriesOptionsType).
+  It's recommended to cast to the specific type for better clarity and
+  correct access to all type-specific options.
```

Comment:
After providing the type property, TypeScript should narrow down to the series type. So no casting is needed. Just set/reset the type in every options object.

Reply:
Even with `type` set it fails when accessing via array index.

[Demo](https://stackblitz.com/edit/highcharts-angular-update-optimal-way-wyejng?file=src%2Fapp%2Fapp.component.ts,package.json).

![image](https://github.com/highcharts/highcharts-documentation-generators/assets/8960408/f9c3dbde-32b3-409e-a3f2-769cb0f6a7db)

Reply:
Okay, I see what you mean. You should do a type check then:

```
if (chart.options.series?.[0].type === 'line') {
    chart.options.series[0].tooltip;
}
```

Reply:
This is a way better way, thank you. Updated.